tensorflow_addons/optimizers/lars_optimizer_test.py

```diff
@@ -0,0 +1,165 @@
+      for dtype in [tf.float16, tf.float32, tf.float64]:
+           shape = [3, 3]
+           var_np = np.ones(shape)
+           grad_np = np.ones(shape)
+           lr_np = 0.1
+           m_np = 0.9
+           wd_np = 0.1
+           ep_np = 1e-5
+           eeta = 0.1
+           vel_np = np.zeros(shape)
+           iterations = 10
+
+           var = tf.Variable(var_np, dtype=dtype)
+           grad = tf.Variable(grad_np, dtype=dtype)
+           opt = lo.LARSOptimizer(
+               learning_rate=lr_np,
+               momentum=m_np,
+               eeta=eeta,
+               weight_decay=wd_np,
+               epsilon=ep_np)
+
+           if not tf.executing_eagerly():
+             self.evaluate(tf.compat.v1.global_variables_initializer())
+             # Fetch params to validate initial values
+             var_tmp = np.ones(shape)
+             self.assertAllClose(var_tmp, self.evaluate(var))
+
+           # initialize the variables for eager mode.
+           if not tf.executing_eagerly():
+             update = opt.apply_gradients([(grad, var)])
+             self.evaluate(tf.compat.v1.global_variables_initializer())
+
+           for _ in range(iterations):
+             if tf.executing_eagerly():
+               opt.apply_gradients([(grad, var)])
+             else:
+               self.evaluate(update)
+
+             post_var = self.evaluate(var)
+
+             # Check we have slots
+             self.assertEqual(["momentum"], opt.get_slot_names())
+             slot0 = opt.get_slot(var, "momentum")
+             self.assertEqual(slot0.get_shape(), var.get_shape())
+             post_vel = self.evaluate(opt.get_slot(var, 'momentum'))
+
+             w_norm = np.linalg.norm(var_np.flatten(), ord=2)
+             g_norm = np.linalg.norm(grad_np.flatten(), ord=2)
+             trust_ratio = eeta * w_norm / (g_norm + wd_np * w_norm + ep_np)
+             scaled_lr = lr_np * trust_ratio
+             grad_np = grad_np + wd_np * var_np
+
+             vel_np = m_np * vel_np + scaled_lr * grad_np
+             var_np -= vel_np
+
+             self.assertAllClose(var_np, post_var, 1e-3, 1e-3)
+             self.assertAllClose(vel_np, post_vel, 1e-3, 1e-3)
+
+  def testGetConfig(self, use_resource=False, use_callable_params=False):
+    lr_np = 0.1
+    m_np = 0.9
+    eeta = 0.1
+    wd_np = 0.1
+    ep_np = 1e-5
+    opt = lo.LARSOptimizer(
+        learning_rate=lr_np,
+        momentum=m_np,
+        eeta=eeta,
+        weight_decay=wd_np,
+        epsilon=ep_np)
+
+    opt = tf.keras.optimizers.deserialize(tf.keras.optimizers.serialize(opt))
+    config = opt.get_config()
+    self.assertEqual(config['learning_rate'], 0.1)
+    self.assertEqual(config['momentum'], 0.9)
+    self.assertEqual(config['weight_decay'], 0.1)
+    self.assertEqual(config['eeta'], 0.1)
+    self.assertEqual(config['epsilon'], 1e-5)
+    self.assertEqual(config['use_nesterov'], False)
+
+
+if __name__ == '__main__':
+  tf.enable_v2_behavior()
```

Comment:
Shouldn't be required as we only test with TF2.

Reply:
done

Reply:
done